util/dnsname/dnsname.go

```diff
@@ -20,14 +20,12 @@ const (
 // A FQDN is a fully-qualified DNS name or name suffix.
 type FQDN string
 
-func ToFQDN(s string) (FQDN, error) {
+// ToFQDNSuffix converts a string to a FQDN that retains any leading '.' in the case of wildcards.
+func ToFQDNSuffix(s string) (FQDN, error) {
```

Comment:
Should this explicitly add a leading `.` if there isn't one? If the user intent is "I want a suffix", then we probably should. We could also make the doc string more exact in this regard - do you always get a suffix matching FQDN, or sometimes you might get just a fully qualified domain name?